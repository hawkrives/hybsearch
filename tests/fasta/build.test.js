/* eslint-env jest */
'use strict'

const expect = require('expect')
const { buildFasta } = require('../../server/lib/fasta/build')

describe('fasta/build', () => {
	it('should generate a fasta file from a fasta object', () => {
		const actual = buildFasta([
			{
				species: 'Emydura_subglobosa__KC755190',
				sequence: 'ggaacaataaattatcacctcaaaagacac',
			},
			{
				species: 'Emydura_victoriae__KC755189',
				sequence: 'ggaacaatcaattattaccccacaaagac',
			},
			{
				species: 'Emydura_victoriae__KC755188',
				sequence: 'ggaacaatcaattattaccccacaaggaa',
			},
		])

		expect(actual).toMatchSnapshot()
	})
})
