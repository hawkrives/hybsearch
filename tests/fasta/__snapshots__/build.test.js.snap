// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fasta/build should generate a fasta file from a fasta object 1`] = `
"> Emydura_subglobosa__KC755190
ggaacaataaattatcacctcaaaagacac
> Emydura_victoriae__KC755189
ggaacaatcaattattaccccacaaagac
> Emydura_victoriae__KC755188
ggaacaatcaattattaccccacaaggaa"
`;
